src/lib/deskflow/ProtocolTypes.h

```diff
@@ -11,365 +11,1046 @@
-extern const char *const kMsgDDragInfo;
-
-// secure input notification:  primary -> secondary
-// $1 = app. app only obtainable on MacOS since that's the only
-// platform facing secure input problems
-extern const char *const kMsgDSecureInputNotification;
-
-// language synchronization:  primary -> secondary
-// $1 = List of server languages
-extern const char *const kMsgDLanguageSynchronisation;
-
-//
-// query codes
-//
-
-// query screen info:  primary -> secondary
-// client should reply with a kMsgDInfo.
-extern const char *const kMsgQInfo;
-
-//
-// error codes
-//
-
-// incompatible versions:  primary -> secondary
-// $1 = major version of primary, $2 = minor version of primary.
-extern const char *const kMsgEIncompatible;
-
-// name provided when connecting is already in use:  primary -> secondary
-extern const char *const kMsgEBusy;
-
-// unknown client:  primary -> secondary
-// name provided when connecting is not in primary's screen
-// configuration map.
-extern const char *const kMsgEUnknown;
-
-// protocol violation:  primary -> secondary
-// primary should disconnect after sending this message.
-extern const char *const kMsgEBad;
-
-//
-// structures
-//
-
-//! Screen information
-/*!
-This class contains information about a screen.
-*/
-class ClientInfo
-{
-public:
-  //! Screen position
-  /*!
-  The position of the upper-left corner of the screen.  This is
-  typically 0,0.
-  */
-  int32_t m_x;
-  int32_t m_y;
-
-  //! Screen size
-  /*!
-  The size of the screen in pixels.
-  */
-  int32_t m_w;
-  int32_t m_h;
+/**
+ * @defgroup protocol_files File Transfer Messages
+ * @brief File transfer and drag-and-drop messages (v1.5+)
+ * @{
+ */
 
-  //! Obsolete (jump zone size)
-  int32_t obsolete1;
+/**
+ * @brief File transfer data
+ *
+ * **Message Code**: `"DFTR"`
+ * **Direction**: Primary ↔ Secondary
+ * **Format**: `"DFTR%1i%s"`
+ * **Parameters**:
+ * - `$1`: Transfer mark (1 byte) - Transfer state
+ * - `$2`: Data (string) - Content depends on mark
+ *
+ * **Transfer Marks**:
+ * - `1` (kDataStart): Data contains file size (8 bytes)
+ * - `2` (kDataChunk): Data contains file content chunk
+ * - `3` (kDataEnd): Transfer complete (data may be empty)
+ *
+ * **Example Transfer Sequence**:
+ * ```
+ * "DFTR\x01\x00\x00\x00\x00\x00\x00\x10\x00"  // Start: 4096 bytes
+ * "DFTR\x02[1024 bytes of file data]"          // Chunk: data
+ * "DFTR\x02[1024 bytes of file data]"          // Chunk: data
+ * "DFTR\x02[1024 bytes of file data]"          // Chunk: data
+ * "DFTR\x02[1024 bytes of file data]"          // Chunk: data
+ * "DFTR\x03"                                   // End: complete
+ * ```
+ *
+ * **Protocol Flow**:
+ * 1. Sender initiates with kDataStart containing total file size
+ * 2. Sender sends multiple kDataChunk messages with file content
+ * 3. Sender concludes with kDataEnd to signal completion
+ * 4. Receiver can abort by closing connection
+ *
+ * @see kMsgDDragInfo, EDataTransfer
+ * @since Protocol version 1.5
```

Comment:
OK, _this_ is the one I meant to deprecate.
```suggestion
 * @since Protocol version 1.5
 * @deprecated File drag and drop is no longer implemented.
```

```cpp

void ClientProxy1_5::fileChunkReceived() const
{
  // do nothing
}

void ClientProxy1_5::dragInfoReceived() const
{
  // do nothing
}
```